turbopack/crates/turbo-tasks/src/task/local_task.rs

```diff
@@ -19,51 +19,39 @@ pub enum LocalTask {
 
 pub fn get_local_task_execution_spec<'a>(
     turbo_tasks: &'_ dyn TurboTasksBackendApi<impl Backend + 'static>,
-    ty: &'a LocalTaskType,
+    ty: &'a LocalTaskSpec,
     // if this is a `LocalTaskType::Resolve*`, we'll spawn another task with this persistence, if
     // this is a `LocalTaskType::Native`, this refers to the parent non-local task.
     persistence: TaskPersistence,
 ) -> TaskExecutionSpec<'a> {
-    match ty {
-        LocalTaskType::Native {
-            native_fn,
-            this,
-            arg,
-        } => {
+    match ty.task_type {
+        LocalTaskType::Native { native_fn } => {
             let span = native_fn.span(TaskPersistence::Local);
             let entered = span.enter();
-            let future = native_fn.execute(*this, &**arg);
+            let future = native_fn.execute(ty.this, &*ty.arg);
             drop(entered);
             TaskExecutionSpec { future, span }
         }
-        LocalTaskType::ResolveNative {
-            native_fn,
-            this,
-            arg,
-        } => {
+        LocalTaskType::ResolveNative { native_fn } => {
             let span = native_fn.resolve_span(TaskPersistence::Local);
             let entered = span.enter();
             let future = Box::pin(LocalTaskType::run_resolve_native(
                 native_fn,
-                *this,
-                &**arg,
+                ty.this,
+                &*ty.arg,
                 persistence,
                 turbo_tasks.pin(),
             ));
             drop(entered);
             TaskExecutionSpec { future, span }
         }
-        LocalTaskType::ResolveTrait {
-            trait_method,
-            this,
-            arg,
-        } => {
+        LocalTaskType::ResolveTrait { trait_method } => {
             let span = trait_method.resolve_span();
             let entered = span.enter();
             let future = Box::pin(LocalTaskType::run_resolve_trait(
                 trait_method,
-                *this,
-                &**arg,
+                ty.this.unwrap(),
```

Comment:
i would say that this is the primary regression.  not in terms of memory or time really just in terms of static type safety